hub/core/compression/tests/test_compression.py

```diff
@@ -0,0 +1,198 @@
+from typing import Union
+import pytest
+import numpy as np
+from hub.core.compression.numpy import NUMPY
+from hub.core.compression.zstd import ZSTD
+from hub.core.compression.lz4 import LZ4
+from hub.core.compression.webp import WEBP
+from hub.core.compression.jpeg import JPEG
+from hub.core.compression.png import PNG
+from hub.core.compression import BaseNumCodec, BaseImgCodec
+
+IMG_ARRAY_SHAPES = (
+    (30, 30, 3),
+    (2, 30, 30, 3),
+    (30, 30, 1),
+)
+GENERIC_ARRAY_SHAPES = ((20, 1), (1, 20), ((60, 60)))
+BENCHMARK_SHAPES = ((30, 30, 3), (5000, 5000, 3))
+ARRAY_DTYPES = (
+    "uint8",
+    "uint16",
+    "float16",
+    "float32",
+)
+LZ4_ACCELERATIONS = (1, 50, 100)
+ZSTD_LEVELS = (1, 11, 22)
+IMG_CODECS = (PNG, JPEG, WEBP)
+NUM_CODECS = (LZ4, ZSTD, NUMPY)
+
+
+def check_equals_decoded(
+    actual_array: np.ndarray, codec: Union[BaseImgCodec, BaseNumCodec]
+) -> None:
+    bytes_ = codec.encode(actual_array)
+    if isinstance(codec, BaseNumCodec):
+        bytes_double = codec.encode(bytes_)
+        bytes_ = codec.decode(bytes_double)
+    decoded_array = codec.decode(bytes_)
+    assert (actual_array == decoded_array).all()
+
+
+def check_codec_config(codec: Union[BaseImgCodec]) -> None:
+    config = codec.get_config()
+    assert config["id"] == codec.__name__
+    assert config["single_channel"] == codec.single_channel
+
+
+def check_codec_single_channel(codec: Union[BaseImgCodec]) -> None:
+    if codec.single_channel:
+        arr = np.ones((100, 100, 1), dtype="uint8")
+    else:
+        arr = np.ones((100, 100), dtype="uint8")
+    check_equals_decoded(arr, codec)
+
+
+@pytest.mark.parametrize("from_config", [False, True])
+@pytest.mark.parametrize("shape", IMG_ARRAY_SHAPES)
```

Comment:
```python
from hub.tests.common import SHAPE_PARAM

FROM_CONFIG_PARAM = "from_config"

parametrize_shape = pytest.mark.parametrize(SHAPE_PARAM, IMG_ARRAY_SHAPES)

...

@parametrize_shape
@pytest.mark.parametrize(FROM_CONFIG_PARAM, [False, True])
test_png_codec(from_config: bool, shape: tuple) -> None:
    ...

```

Reference:
https://github.com/activeloopai/Hub/blob/1e7e454ff1ebc49d8d1f989424540c3d9eb0204c/hub/tests/common.py#L12

Reply:
then any tests that do `@pytest.mark.parametrize("shape", IMG_ARRAY_SHAPES)` can simply use `@parametrize_shape` instead